compiler/bindings/python/iree/compiler/api/ctypes_dl.py

```diff
@@ -338,8 +338,13 @@ def wrap_buffer(
                 buffer_name.encode(),
                 buffer,
                 buffer_len,
-                # Detect if nul terminated.
-                True if buffer_len > 0 and view[-1] == 0 else False,
+                # Detect if null terminated.
+                # MLIR bytecode files are not null terminated, but could be by chance.
+                (
+                    True
+                    if buffer_len > 0 and view[-1] == 0 and view[:4].hex() != "4d4cef52"
+                    else False
+                ),
```

Comment:
I'm not following here. Why was the check for the final character not sufficient? Why also check for `MLIR` (if I'm to trust this hex to ascii converter I found :P)?

Reply:
I'd need to look a bit closer at what the issue is, but my feeling is that we should keep the "is null terminated" check separate from the "is MLIR bytecode" check. Could move that logic up or down into these functions as needed.

Reply:
Yeah, let's discuss that in the follow up PR then. 👍